parser/shared/src/main/scala/hmda/parser/fi/lar/LarCsvParser.scala

```diff
@@ -145,16 +145,49 @@ object LarCsvParser {
         "Lien Status" -> fields(38)
       )
 
-      val validationList = numericFields.map { case (key, value) => toIntOrFail(value, key) }
+      val NAFields = ListMap(
+        "Date Application Received" -> fields(4),
+        "MSA" -> fields(13),
+        "State" -> fields(14),
+        "County" -> fields(15),
+        "Census Tract" -> fields(16),
+        "Rate Spread" -> fields(36)
+      )
+
+      val numericValidationList = numericFields.map { case (key, value) => toIntOrFail(value, key) }
+      val naValidationList = NAFields.map { case (key, value) => checkNA(value, key) }
+      val validationList = numericValidationList ++ naValidationList
 
       validationList.reduce(_ +++ _)
     }
   }
 
   def toIntOrFail(value: String, fieldName: String): ValidationNel[String, List[Int]] = {
     Try(value.toInt) match {
-      case Failure(result) => s"$fieldName is not an integer".failure.toValidationNel
+      case Failure(_) => s"$fieldName is not an integer".failure.toValidationNel
       case Success(result) => List(result).success
     }
   }
+
+  def checkNA(value: String, fieldName: String): ValidationNel[String, List[Int]] = {
+    if (value == "") {
+      s"$fieldName cannot have empty value".failure.toValidationNel
+    } else if (value == "NA") {
+      List(0).success
+    } else {
+      val dot = value.indexOf(".")
```

Comment:
This if-else logic could be simplified to `value.replaceFirst(raw"\.", "")`, while still making sure that a string like "1.2.3" will fail.